mongodbatlas/fw_resource_mongodbatlas_search_deployment.go

```diff
@@ -260,14 +277,15 @@ func splitSearchNodeImportID(id string) (projectID, clusterName string, err erro
 	return
 }
 
-func waitSearchNodeStateTransition(ctx context.Context, projectID, clusterName string, client *admin.APIClient, timeout time.Duration) (*admin.ApiSearchDeploymentResponse, error) {
+func waitSearchNodeStateTransition(
```

Comment:
Having a separate file for this function, and its corresponding _unit.go file would be ideal here. Having more granular files will can be enabled if we separate implementation into packages (https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1643)

Reply:
yes! I wonder if we can move forward with that change